subsys/bluetooth/host/conn.c

```diff
@@ -1114,13 +1114,27 @@ static void add_pending_tx(struct bt_conn *conn, bt_conn_tx_cb_t cb)
 	key = irq_lock();
 	sys_slist_append(&conn->tx_pending, node);
 	irq_unlock(key);
+
+	return node;
+}
+
+static void remove_pending_tx(struct bt_conn *conn, sys_snode_t *node)
+{
+	unsigned int key;
+
+	key = irq_lock();
```

Comment:
We should just change the driver->send() errors to asserts, which will make the need to remove from the list in this place go away.